core/lib/instrument_api.h

```diff
@@ -1403,6 +1403,23 @@ bool dr_unregister_signal_event(dr_signal_action_t (*func)(void *drcontext,
 #    endif /* UNIX */
 /* DR_API EXPORT END */
 
+DR_API
+/**
+ * Registers a callback function for the low on memory event.  DR calls \p func
+ * whenever virtual memory is tight and enables the client to help free space.
+ */
+void
+dr_register_low_on_memory_event(void (*func)(void *drcontext));
+
+DR_API
+/**
+ * Unregister a callback function for low on memory events.
+ * \return true if unregistration is successful and false if it is not
+ * (e.g., the function was not registered).
+ */
+bool
+dr_unregister_low_on_memory_event(void (*func)(void *drcontext));
+
 /****************************************************************************
  * SECURITY SUPPORT
  */
```

Comment:
Please add tests:

1. At least add calls to reg & unreg to suite/tests/client-interface/events.dll.c
2. Ideally, add some kind of targeted test where this event is triggered.  Testing OOM is not always easy.  How about a test which passes -vm_size 4M or sthg, which should be easy to fill up?

Reply:
I added the reg & unreg tests. Is it okay with you if I add the OOM test when I add the drmgr wrapper? Other than that, I think I am done.

Reply:
On second thought, I think it's best I add that test now, even though users won't actually call that function directly in the future. 

I'll see what I can do.

Reply:
SGTM.

Reply:
Is it possible to set vm_size via code instead of a runtime param?

Reply:
No, but I'm not sure I understand the reason?  It's a startup param so you'd need separate runs for each variation in any case.  The option would be set in suite/tests/CMakeLists.txt: each test has its own parameters there, so it seems straightforward to set it there?

Reply:
I included the test